pygsti/algorithms/grasp.py

```diff
@@ -67,8 +66,15 @@ def neighboring_weight_vectors(weights, forced_weights=None, shuffle=False):
     return neighbors
 
 
-def _grasp_construct_feasible_solution(elements, score_fn, rcl_fn, feasible_threshold=None,
-                                       feasible_fn=None, initial_elements=None):
+def _grasp_construct_feasible_solution(
+    elements,
+    score_fn,
+    rcl_fn,
+    feasible_threshold=None,
+    feasible_fn=None,
+    initial_elements=None,
+    rng=None,
```

Comment:
Marked as substantial change - adds rng arg to signature, docstring, and within function body below.